### DISCUSSION TITLE
Tuple (for lack of a better name)

### DISCUSSION THREAD

**Discussion by skttl** [leekelleher/umbraco-contentment#337] [Idea]
For creating values of eg <IPublishedContent, string>, a tuple editor is a wrapper for 2 or more existing datatype. The datatypes could in theory be anything, like a block grid and block list (good luck finding a use case for that 😜). 

In regards to UI my idea is to make a flex ox with wrapping enabled (could be configurable) , and then just pop the editors in next to each other.

My use case right now is to make a text string editor, with a dropdown next to it, to select heading level.

The value converter should simply return a tuple with each of the datatypes return values.

**Comment by leekelleher** [comment]
For reference, I did similar for Umbraco 7, but never released it, called Tuple: https://github.com/umco/umbraco-tuple

It could work in combination with the [Property List](https://github.com/umco/umbraco-property-list) editor for v7.

---

I did have a random idea that this could be done as a custom Display Mode for the Content Blocks editor, e.g. an inline view for editing a small set of fields on an element type. I'm not sure if it'd be overkill, but would reduce the amount of code needed to configure the tuple, in terms of DataType and ModelsBuilder.  I'll give it some thought.  (Although any new development on existing Contentment is already sending shivers down my spine about porting it over for v14/Bellissima 😅)

**Comment by leekelleher** [comment]
Closing this off, as the feature will be part of **Input List** for Contentment v7.0. #513
